jaeger-context/src/main/java/com/uber/jaeger/context/TracingUtils.java

```diff
@@ -18,9 +18,18 @@
 
 import java.util.concurrent.ExecutorService;
 
+/**
+ * TracingUtils is going to be deprecated. To use its method please provide a tracer via
+ * {@link com.uber.jaeger.context.TracingUtils#setTracer(io.opentracing.Tracer)}.
+ */
+@Deprecated
 public class TracingUtils {
   private static TraceContext traceContext = new ActiveSpanSourceTraceContext(GlobalTracer.get());
 
+  public static void setTracer(io.opentracing.Tracer tracer) {
+    GlobalTracer.register(tracer);
```

Comment:
This is not quite what I meant. Rather than depending on global tracer AT ALL, this could simply store the tracer and init traceContext with it. The other static methods check for traceContext to be initialized, if it's not they throw a descriptive error saying that `setTracer` must be called before using this class.

Reply:
Gotcha!